gcovr/utils.py

```diff
@@ -69,6 +69,29 @@ def realpath(path):
     realpath = os.path.realpath
 
 
+def is_fs_case_insensitive():
+    cwd = os.getcwd()
+    # Guessing if file system is case insensitive.
+    # The working directory is not the root and accessible in upper and lower case.
+    return (
+        (cwd != os.path.sep)
+        and os.path.exists(cwd.upper())
+        and os.path.exists(cwd.lower())
+    )
+
+
+def resolvePathCaseStyle(path):
+    if (os.path.dirname(path) == path):
+        return path
+
+    if os.path.islink(path):
+        matchedFileName = [f for f in os.listdir(os.path.dirname(path)) if f.casefold() == os.path.basename(path).casefold()][0]
+    else:
+        matchedFileName = os.path.basename(os.path.realpath(path))
+
+    return os.path.join(resolvePathCaseStyle(os.path.dirname(path)), matchedFileName)
```

Comment:
I think `casefold` is wrong here since it converts e.g. `ß` to `ss` which would give a wrong result if two files differ in this letter.
The different handling of symlinks can't be tested in an easy way, therefore I prefer to always use the same code to get the correct case.

Since the file system is queried for each part of each file path this function should use a cache.

Untested:
```suggestion
@functools.lru_cache(maxsize=None)
def resolvePathCaseStyle(path: str):
    if (os.path.dirname(path) == path):
        return path

    matching_files = glob.glob(os.path.basename(path), root_dir=os.path.dirname(path)
    assert len(matching_files) == 1  # Seems that we have a case sensitve filesystem
    name = matching_files[0]

    return os.path.join(resolvePathCaseStyle(os.path.dirname(path)), name)
```

Reply:
> I've some suggestions. Is it possible to write a test which triggers the change of the case?

Unfurtunately it has shown that reproducing the error is not that easy. It requires following steps: 
1. The source code has to contain definitions in header files
2. The tests are compiled on Windows using a cross compiler
3. The tests are executed on the target and the .gcda files are uploaded to the windows machine
4. gcovr is used to evaluates the code coverage  on the windows machine. 

I would suggest to wirte a UnitTest for resolvePathCaseStyle()